File: ballast-core/npm/src/domain/events/player-signed-in.ts
import { EventBase } from "../../messaging/event-base";
import { IDomainEvent } from "../domain-event";
import { Player } from "../models/player";

export class PlayerSignedInDomainEvent extends EventBase implements IDomainEvent {

    public static readonly id: string = "PlayerSignedInDomainEvent";
    public get id() {
        return PlayerSignedInDomainEvent.id;
    }

    public readonly player: Player; 

    private constructor(eventDateIsoString: string, player: Player) {
        super(eventDateIsoString);
        this.player = player;
    }

    public static fromPlayer(player: Player) {
        return new PlayerSignedInDomainEvent(
            EventBase.getDateIsoString(),
            player
        );
    }

}

File: ballast-core/dotnet/src/application/events/PlayerSignedIn.cs
using Ballast.Core.Application.Models;
using Ballast.Core.Messaging;

namespace Ballast.Core.Application.Events
{
    public class PlayerSignedInEvent : EventBase, IApplicationEvent  
    {

        public static string GetId() => nameof(PlayerSignedInEvent);
        public override string Id => GetId();

        public PlayerDto Player { get; private set; }

        private PlayerSignedInEvent() { } // For model binding / deserialization
        private PlayerSignedInEvent(string eventDateIsoString, PlayerDto player) : base(eventDateIsoString) 
        {
            Player = player;
        }

        public static PlayerSignedInEvent FromPlayer(PlayerDto player) =>
            new PlayerSignedInEvent(
                EventBase.GetDateIsoString(),
                player
            );

    }
}